### PR TITLE
Navigation: Reset submenu before making Flyout

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/woocommerce-admin/issues/6377

Some automatically migrated menu items are still included in the WooCommerce Submenu, so they are also added to the flyout. This PR makes sure all items have the `hide-if-js` before constructing the flyout.

@joshuatf This works, but I'm not sure its the best way. I'm not fully understanding how MailChimp gets migrated, but it seems that would be a better time to add the `hide-if-js` class than what I've done. Also, just to note this solution results in the class added twice if its already present, which I'll fixup if this is the direction.

### Detailed test instructions:

1. Activate the MailChimp plugin
2. Turn on Navigation
3. See the flyout on the WP screen and MailChimp should not be included.

### CODE REVIEW

**Review by joshuatf — COMMENTED** [review]
Nice fix!  This is testing well for me.

My only comment is that this should probably be moved into the `migrate_core_child_items` method in `Menu.php`.  

This might need a couple small changes to work as `$submenu_items` merges two arrays and loses the original keys.  Maybe we just loop over `$submenu['woocommerce']` and the product item separately.

**Comment by psealock** [comment]
Thanks for the review @joshuatf 

I moved the function to `migrate_core_child_items` and split out the loop innards into a separate function so product items and woo main items can be iterated over separately.

File: src/Features/Navigation/Menu.php [hunk 1]
```diff
@@ -513,6 +513,54 @@ public function add_core_items() {
 		}
 	}
 
+	/**
+	 * Add an item or taxonomy.
+	 *
+	 * @param array $menu_item Menu item.
+	 */
+	public function add_item_and_taxonomy( $menu_item ) {
+		if ( in_array( $menu_item[2], CoreMenu::get_excluded_items(), true ) ) {
```
Comment (joelclimbsthings) [inline src/Features/Navigation/Menu.php:522]: You reference `$menu_item[2]` several times here. Totally optional, but I might assign this to a temporary variable with a descriptive name.

**Review by joelclimbsthings — APPROVED** [review]
Testing well, and code looks good. I added one entirely optional note. :ship:

**Pull request merged**